notebooks/community/aisamples/AIsample - Book Recommendation.ipynb

```diff
@@ -1254,9 +1259,60 @@
    },
    "outputs": [],
    "source": [
-    "saved_model_path = f\"{DATA_FOLDER}/als_model/{int(time.time())}\"\n",
-    "model.save(saved_model_path)\n",
-    "print(f\"model is saved in: {saved_model_path}\")"
+    "# setup mlflow\n",
+    "import mlflow\n",
+    "import trident.mlflow\n",
+    "from trident.mlflow import get_sds_url\n",
+    "\n",
+    "mlflow.set_tracking_uri(get_sds_url())\n",
+    "mlflow.set_registry_uri(get_sds_url())\n",
+    "mlflow.set_experiment(EXPERIMENT_NAME)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# log model, metrics and params\n",
+    "with mlflow.start_run() as run:\n",
+    "    print(\"log model:\")\n",
+    "    mlflow.spark.log_model(\n",
+    "        model,\n",
+    "        f\"{EXPERIMENT_NAME}-alsmodel\",\n",
+    "        registered_model_name=f\"{EXPERIMENT_NAME}-alsmodel\",\n",
+    "        dfs_tmpdir=\"Files/spark\",\n",
+    "    )\n",
+    "\n",
+    "    print(\"log metrics:\")\n",
+    "    mlflow.log_metrics({\"RMSE\": rmse, \"MAE\": mae, \"R2\": r2, \"Explained variance\": var})\n",
+    "\n",
+    "    print(\"log parameters:\")\n",
+    "    mlflow.log_params(\n",
+    "        {\n",
+    "            \"num_epochs\": num_epochs,\n",
+    "            \"rank_size_list\": rank_size_list,\n",
+    "            \"reg_param_list\": reg_param_list,\n",
+    "            \"model_tuning_method\": model_tuning_method,\n",
+    "            \"DATA_FOLDER\": DATA_FOLDER,\n",
+    "        }\n",
+    "    )\n",
+    "\n",
+    "    model_uri = f\"runs:/{run.info.run_id}/{EXPERIMENT_NAME}-alsmodel\"\n",
```

Comment:
Can this be simplified with @serena-ruan 's Autologging?

Reply:
Currently `mlflow.pyspark.ml.autolog` doesn't work since the default `dfs_tmpdir` is `/tmp/mlflow`, which doesn't work in our platform. Need to wait for another release of mlflow which will include Serena's PR for setting `DFS_TMP` as an environment variable. Moreover, for ALS model, it's not supported by autolog. I see Serena has two PRs for supporting modifying `logModelAllowlistFile` in a user friendly way, I guess those will be released in the next version of mlflow too.